### DISCUSSION TITLE
`ImGui::IsKeyDown('M')` to `ImGui::IsKeyDown(static_cast<ImGuiKey>('M'))`?

### DISCUSSION THREAD

**Discussion by DominiquePelle-TomTom** [ocornut/imgui#5933] [New Users Build/Link/Run issues ONLY!]
My code has things like this `ImGui::IsKeyDown('M')` which does
not compile when trying to upgrade to imgui-1.89.1.  Looking at the
imgui code, I see this comment in `imgui.h:1353`:
```
  // A key identifier (ImGuiKey_XXX or ImGuiMod_XXX value)
  // All our named keys are >= 512. Keys value 0 to 511 are left unused as legacy native/opaque key values (< 1.87)
  // Since >= 1.89 we increased typing (went from int to enum), some legacy code may need a cast to ImGuiKey.
  enum ImGuiKey : int
```

The above comment suggests to cast, and I can see that changing code
to `ImGui::IsKeyDown(static_cast<ImGuiKey>('M'))` (i.e. cast 'M' to an
enum) makes it build, but is this correct?  Even if it works, is there a
cleaner way since the comment indicate that "**legacy** code may need
to cast".

**Comment by ocornut** [comment]
Using ‘M’ has always been backend-dependant.

Using the same backend you used you can cast as ImGuiKey and it’ll work. But you may use ImGuiKey_M and it’ll be portable.

See:
https://github.com/ocornut/imgui/issues/4921

I might have made a mistake increasing typing safety for ImGuiKey as this casting is required for legacy paths.
